<launch>

    <arg name="enable_debug"    default="false"/>
    <arg name="input_cloud_topic" default="/points"/>
    <arg name="main_cloud_topic" default="/main_cloud"/>
    <arg name="cost_cloud_topic" default="/cost_cloud"/>
    <arg name="cost_norm_cloud_topic"  default="/cost_norm_cloud"/>
    <arg name="fixed_frame_id" default="map"/>
    <!-- <arg name="enable_rviz" default="true"/> -->
    <!-- <arg name="pub_viz" default="true"/> -->

    <arg name="config_filepath"     default="$(find curvy_terrain_mapper)/cfg/config.yaml"/>

    <arg name="launch_prefix" default="gdb -ex run --args" if="$(eval arg('enable_debug'))"/>
    <arg name="launch_prefix" default="" unless="$(eval arg('enable_debug'))"/>

    <node pkg="curvy_terrain_mapper" type="curvy_terrain_mapper_node" name="curvy_terrain_mapper" output="log" respawn="true" launch-prefix="$(arg launch_prefix)">
        <remap from="~input_cloud" to="$(arg input_cloud_topic)"/>
        <remap from="~main_cloud" to="$(arg main_cloud_topic)"/>
        <remap from="~cost_cloud" to="$(arg cost_cloud_topic)"/>
        <remap from="~cost_norm_cloud" to="$(arg cost_norm_cloud_topic)"/>
        <param name="~fixed_frame_id" value="$(arg fixed_frame_id)"/>
        <rosparam command="load" file="$(arg config_filepath)"/>
    </node>
</launch>
